packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -65,56 +71,69 @@ const shouldApplyFilterItemOnGroup = (columnField: string, node: GridRowTreeNode
 export const filterRowTreeFromGroupingColumns = (
   params: FilterRowTreeFromTreeDataParams,
 ): Omit<GridFilterState, 'filterModel'> => {
-  const { rowTree, isRowMatchingFilters } = params;
+  const { rowTree, isRowMatchingFilters, filterModel } = params;
   const visibleRowsLookup: Record<GridRowId, boolean> = {};
   const filteredRowsLookup: Record<GridRowId, boolean> = {};
   const filteredDescendantCountLookup: Record<GridRowId, number> = {};
 
   const filterTreeNode = (
     node: GridRowTreeNodeConfig,
-    areAncestorsPassingChildren: boolean,
```

Comment:
The `areAncestorsPassingChildren` is not needed anymore because now to compute the filtering we do not have early return